azurecontainerapps.ts

```diff
@@ -232,13 +232,43 @@ export class azurecontainerapps {
     private static async getLocation(): Promise<string> {
         // Set deployment location, if provided
         let location: string = this.toolHelper.getInput('location', false);
+        let resourceGroup: string = this.toolHelper.getInput('resourceGroup', false);
+        let containerAppEnvironment: string = this.toolHelper.getInput('containerAppEnvironment', false);
+
+        if (!this.util.isNullOrEmpty(location)) {
+            return location;
+        }
 
-        // If no location was provided, use the default location for the Container App service
-        if (this.util.isNullOrEmpty(location)) {
-            location = await this.appHelper.getDefaultContainerAppLocation();
+        // If no location was provided, attempt to discover the location of the existing Container App Environment linked to the Container App
+        // or Container App Environment provided in the resource group or use the default location.
+
+        if (!this.util.isNullOrEmpty(resourceGroup)) {
+            // Check if Container App exists in the resource group provided and get the location from the Container App Environment linked to it
+            let doesContainerAppExist = await this.appHelper.doesContainerAppExist(this.containerAppName, resourceGroup);
```

Comment:
There's an existing global variable `containerAppExists` that we should set the value of here. Anywhere else in the code that this variable is used we should check to see if its value is already set to avoid making an additional Azure CLI call to verify if the Container App exists or not.

Reply:
Updated.